app/components/DepositWithdraw/WithdrawModalNew.jsx

```diff
@@ -0,0 +1,834 @@
+        let p = [];
+        assets.forEach(a => {
+            p.push(checkFeeStatusAsync({
+                accountID: account.get("id"),
+                feeID: a,
+                options: ["price_per_kbyte"],
+                data: {
+                    type: "memo",
+                    content: state.selectedAsset.toLowerCase() + ":" + state.address + (state.memo ? ":" + state.memo : "")
+                }
+            }));
+        });
+        Promise.all(p).then(status => {
+            assets.forEach((a, idx) => {
+                feeStatus[a] = status[idx];
+            });
+            if (!utils.are_equal_shallow(state.feeStatus, feeStatus)) {
+                this.setState({
+                    feeStatus
+                });
+            }
+        }).catch(err => {
+            console.error(err);
+        });
+    }
+
+    _updateFee(state = this.state) {
+        let btsAccount = this.props.account;
+        let { fee_asset_id } = state;
+        const { fee_asset_types } = this._getAvailableAssets(state);
+        if ( fee_asset_types.length === 1 && fee_asset_types[0] !== fee_asset_id) {
+            fee_asset_id = fee_asset_types[0];
+        }
+
+        if (!btsAccount) return null;
+        checkFeeStatusAsync({
+            accountID: btsAccount.get("id"),
+            feeID: fee_asset_id,
+            options: ["price_per_kbyte"],
+            data: {
+                type: "memo",
+                content: this.props.output_coin_type + ":" + state.withdraw_address + (state.memo ? ":" + state.memo : "")
+            }
+        })
+        .then(({fee, hasBalance, hasPoolBalance}) => {
+            if (this.unMounted) return;
+
+            this.setState({
+                feeAmount: fee,
+                hasBalance,
+                hasPoolBalance,
+                feeError: (!hasBalance || !hasPoolBalance)
+            });
+        });
+    }
+
+    _getStyleHelpers(){
+        let halfWidth = {width: "50%", float: "left", boxSizing: "border-box"}
+        let leftColumn = _.extend({paddingRight: "0.5em", marginBottom: "1em"}, halfWidth);
+        let rightColumn = _.extend({paddingLeft: "0.5em", marginBottom: "1em"}, halfWidth);
+        let buttonStyle = {width: "100%"}
+
+        return { halfWidth, leftColumn, rightColumn, buttonStyle }
+    }
+
+    _getBindingHelpers(){
+        let onFocus = this.onFocusAmount.bind(this);
+        let onBlur = this.onBlurAmount.bind(this);
+
+        return { onFocus, onBlur }
+    }
+
+    onFeeChanged({asset}) {
+        this.setState({
+            fee_asset_id: asset.get("id")
+        }, this._updateFee);
+    }
+
+    onAssetSelected(value, asset){
+        let { selectedAsset, selectedGateway } = _onAssetSelected.call(this, value);
+        let address = WithdrawAddresses.getLast(value.toLowerCase());
+        this.setState({selectedAsset, selectedGateway, gateFee: asset.gateFee, issuer: asset.issuer, address, isBTS: false});
+    }
+
+    onAssetChanged(value){
```

Comment:
Should clear all input fields as we're switching asset. Clear Amount, Address and Memo

Reply:
Check